src/runtime/critical_cortexm.go

```diff
@@ -0,0 +1,22 @@
+// +build cortexm
+
+package runtime
+
+import "device/arm"
+
+var criticalLevel uint32
+var criticalState uintptr
+
+func critical() {
+	if criticalLevel == 0 {
+		criticalState = arm.DisableInterrupts()
```

Comment:
There is a much better way to do this, and it is with nested interrupts:

    mask := arm.DisableInterrupts()
    // critical section
    arm.EnableInterrupts(mask)

It is safe to use this pattern in a nested fashion. Note that `EnableInterrupts` is a bit of a misnomer, maybe it should be called `RestoreInterrupts` as it restores the interrupt-enabled state from before the `DisableInterrupts` call.

The added benefit is that the code is much smaller as there are no globals involved.

Reply:
I was trying to make something that would work also on platforms where interrupt disabling worked differently, although now i guess for them we can use a global and return a boolean.